Fix sourcemaps with traceur 0.0.72

guzart opened the pull request:
Fixes #68 

@thlorenz I am not sure what's the benefit of rebuilding the sourcemap to "Ignore mappings that are not from our source file". Could you please explain? Thanks.

guzart committed:
Fix sourcemaps with traceur 0.0.72

Fixes #68

domenic commented on compile.js:
```diff
@@ -9,25 +9,34 @@ var traceurOptions = {
   }
+
+  if (options.sourceMaps === true) {
+    options.sourceMaps = 'inline';
```
Isn't the problem that people are passing `true` when they should be passing `'inline'`? Should we be allowing them to pass wrong options? Seems bad.

guzart replied:
I guess I was trying to go for backward compatibility. But also because this is a Browserify plugin, we need to have some control over how the sourcemaps are generated, so that we can pass them to Browserify the way its being expected.

guzart replied:
I see your point. Having an invalid traceur option value in a "traceurOverrides" property seems bad indeed.

domenic replied:
Although we apparently already do it for `sourceMap`, singular. Perhaps add a similar deprecation warning, then do a 1.0.0 release that removes them.

domenic commented on compile.js:
```diff
@@ -9,25 +9,34 @@ var traceurOptions = {
 
-    var result = compiler.compile(contents, file);
-    var sourceMap = compiler.getSourceMap();
+    var result = compiler.compile(contents, file, file + '.es6');
```
Why `.es6`? That is not the standard extension for ES6 files. (`.js` is.)

guzart replied:
Was trying to preserve the code from [index.js](https://github.com/thlorenz/es6ify/blob/master/index.js#L36). But I don't understand why it's needed either, probably unnecessary.

domenic commented:
Would be good to get @thlorenz's thoughts on this. This does look a lot more sane and less error-prone than what we are doing now but I imagine there was a reason for that so...

thlorenz commented:
I'm not sure I remember the reason for `es6` extension. It may have been to distinguish the compiled file (`.js`) from the original (`.es6`).
However if everything is working and all tests pass I have no objections to the changes made here.

I'm also ok with breaking backwards compat when bumping the major version and having deprecate warning in the meantime.

guzart committed:
Update transform test to compare against traceur generated sourcemap

guzart committed:
Add deprecation warning for traceurOverrides.sourceMap = true

guzart commented:
I have:
- Updated the sourcemap transform test to consider the traceur additions
- Updated convert-source-map depdency, because the previous version could not convert the traceur sourcemap.
- Added the deprecation warning when `traceurOverrides.sourceMaps = true`

Tests are `ok`

thlorenz commented:
LGTM, please merge into master. Either @domenic or I will publish a new version.
Thanks.

domenic commented:
Thanks, merged as 5ff0da929c89cf2289ae4a24490540405ba0688c
